OpenSim/Simulation/Model/ModelComponent.cpp

```diff
@@ -76,7 +79,16 @@ void ModelComponent::extendConnect(Component &root)
         connectToModel(*model);
 }
 
-
+void ModelComponent::extendFinalizeFromProperties()
+{
+    Super::extendFinalizeFromProperties();
+    int geomSize = getProperty_GeometrySet().size();
+    if (geomSize > 0){
+        for (int i = 0; i < geomSize; ++i){
+            addComponent(&upd_GeometrySet(i));
```

Comment:
@sherm1 `MoelComponent::finalizeFromProperties()` is not being called from the constructor nor should it be. This is the correct place to be doing `addComponent()` as well.